### PR TITLE
Add nRF52840DK-based Thread tutorial board definition

### PR DESCRIPTION
**Pull request by lschuermann**
### Pull Request Overview

This pull request adds a board definition for the nRR52840DK board we'll be using for the Thread tutorial at CPS-IoT Week. It's still a draft as we might need to add more things / split things out into other PRs.

### Testing Strategy

Running the tutorial apps.

### TODO or Help Wanted

- [ ] Documentation for the board.
- [x] Should be moved to `boards/tutorials`.
- [x] Split out the `lib/bin` crate refactor in the `nrf52840dk` crate into its own PR?

### Documentation Updated

- [ ] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [ ] Ran `make prepush`.

### CODE REVIEW

**Comment by lschuermann** [comment]
> Can we merge the bin/lib split first?

Yup, working on that right now!

**Comment by lschuermann** [comment]
@bradjc Implemented in #3981, I'll rebase this PR on that.

**Comment by lschuermann** [comment]
This should be ready as soon as CI passes.

File: boards/tutorials/nrf52840dk-thread-tutorial/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+name = "nrf52840dk-thread-tutorial"
+version.workspace = true
+authors.workspace = true
+build = "../../build.rs"
+edition.workspace = true
+
+[features]
+default = ["screen_ssd1306"]
+screen_ssd1306 = []
+screen_sh1106 = []
```
Comment (bradjc) [inline boards/tutorials/nrf52840dk-thread-tutorial/Cargo.toml:15]: Cargo features just do not work for us do they.

**Review by bradjc — APPROVED** [review]

File: boards/nordic/nrf52840dk/src/lib.rs [hunk 2]
```diff
@@ -418,8 +416,13 @@ pub unsafe fn start() -> (
...
             7 => &nrf52840_peripherals.gpio_port[Pin::P1_08],
-            8 => &nrf52840_peripherals.gpio_port[Pin::P1_10],
-            9 => &nrf52840_peripherals.gpio_port[Pin::P1_11],
+            // Avoid exposing the I2C pins to userspace, as these are used in
+            // some tutorials (e.g., `nrf52840dk-thread-tutorial`).
+            //
+            // In the future we might want to make this configurable.
+            //
+            // 8 => &nrf52840_peripherals.gpio_port[Pin::P1_10],
+            // 9 => &nrf52840_peripherals.gpio_port[Pin::P1_11],
```
[the remark this answers is not in the document]
Comment (lschuermann) [reply boards/nordic/nrf52840dk/src/lib.rs:425]: Yeah...

File: boards/tutorials/nrf52840dk-thread-tutorial/Cargo.toml (hunk 1 above)
Comment (lschuermann) [reply boards/tutorials/nrf52840dk-thread-tutorial/Cargo.toml:15]: :confused: I know, do you feel like we should remove this? Really this is only such that we can still easily support users with the wrong model / develop using the other screen that we have.

**Comment by lschuermann** [comment]
Can we merge this today? I want to rebuild the VM image with this PR landed some time tomorrow.

**Comment by lschuermann** [comment]
Merging as we need this for the tutorial, if this breaks anything we should feel free to revert afterwards.

**Pull request merged**